### PR TITLE
Bazefetcher source

### PR DESCRIPTION
**Pull request by JensGM**

### CODE REVIEW

File: camille/source/bazefetcher.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    #     |- start_date ----------| |- end_date ------------|
+    # tag_YYYY-MM-DDTHH.MM.SS+HH.MM_YYYY-MM-DDTHH.MM.SS+HH.MM.json.gz
+    date_str = fn[len(tag)+1:-34] # Slice out the start date
+    date_str = date_str.replace('.', '') # UTC offset does not support '.'
+    date_fmt = "%Y-%m-%dT%H%M%S%z"
+    return datetime.datetime.strptime(date_str, date_fmt)
+
+
+def _read_json_gz(fn):
+    with gzip.GzipFile(fn, 'r') as f:
```
Comment (jokva) [inline camille/source/bazefetcher.py]: Does this crash if fn does not exist?

File: camille/source/bazefetcher.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        return pd.DataFrame(json.load(f))
+    return pd.DataFrame()
+
+
+def bazefetcher(root, tzinfo=pytz.utc):
+    if not os.path.isdir(root):
+        raise ValueError('{} is not a directory'.format(root))
+
+    if not isinstance(tzinfo, datetime.tzinfo):
+        raise ValueError('tzinfo must be instance of datetime.tzinfo')
```
Comment (jokva) [inline camille/source/bazefetcher.py]: Maybe let pandas itself figure out if it accepts tzinfo or not.

File: camille/source/bazefetcher.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    def bazefetcher_internal(tag, start_date, end_date):
+        if (not isinstance(start_date, datetime.date)
+            or not isinstance(end_date, datetime.date)):
+            raise ValueError(
+                'start_date and end_date parameters must be of type date')
+
+        if not start_date.tzinfo or not end_date.tzinfo:
```
Comment (jokva) [inline camille/source/bazefetcher.py]: `is not None`

File: tests/test_data/synthetic_bazefield_generator.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
```
Comment (jokva) [inline tests/test_data/synthetic_bazefield_generator.py]: `env python3`

File: camille/source/bazefetcher.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import gzip
+import pytz
+import pandas as pd
+
+
+def _fn_start_date(tag, fn):
+    # File names are on the form:
+    #     |- start_date ----------| |- end_date ------------|
+    # tag_YYYY-MM-DDTHH.MM.SS+HH.MM_YYYY-MM-DDTHH.MM.SS+HH.MM.json.gz
+    date_str = fn[len(tag)+1:-34] # Slice out the start date
```
Comment (sveinung-r) [inline camille/source/bazefetcher.py]: Perhaps better to split on underscore?

File: tests/source/test_bazefetcher.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+end_date = datetime.datetime(2018, 5, 11, tzinfo=pytz.utc)
+start_date_no_tz = datetime.datetime(2018, 4, 9)
+future_date = datetime.datetime(2018, 6, 1, tzinfo=pytz.utc)
+bazefield = bazefetcher('tests/test_data/bazefield_data_root')
+
+
+def test_bazefetcher():
+    tag2 = bazefield('installation-04-tag-2', start_date, end_date)
+    assert (1.100000e+32 < tag2).all()
+    assert (tag2 < 1.300000e+32).all()
```
Comment (sveinung-r) [inline tests/source/test_bazefetcher.py]: Could also check count

**Pull request merged**